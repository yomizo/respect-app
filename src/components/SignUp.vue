<template>
  <v-content class="grey darken-3">
    <v-container class="formx">
      <v-card>
        <v-container>
          <div class="headline font-weight-bold">SignUp</div>
          <form>
            <v-text-field
              v-model="name"
              v-validate="'required|max:10'"
              :counter="10"
              :error-messages="errors.collect('name')"
              label="Name"
              data-vv-name="name"
              required
            ></v-text-field>
            <v-text-field
              v-model="email"
              v-validate="'required|email'"
              :error-messages="errors.collect('email')"
              label="E-mail"
              data-vv-name="email"
              required
            ></v-text-field>
            <v-text-field 
              v-model="password"
              v-validate="'required'" 
              name="password" 
              type="password" 
              :class="{'is-danger': errors.has('password')}" 
              placeholder="Password" 
              ref="password"></v-text-field>
            <div class="text-xs-right">
              <v-btn @click="submit">submit</v-btn>
              <v-btn @click="clear">clear</v-btn>
            </div>
          </form>
        </v-container>
      </v-card>
    </v-container>
  </v-content>
</template>

<script>
  import Vue from 'vue'
  import VeeValidate from 'vee-validate'

  Vue.use(VeeValidate)

  export default {
    $_veeValidate: {
      validator: 'new'
    },

    data: () => ({
      name: '',
      email: '',
      password: '',
      // flash_message_dic
      dictionary: {
        attributes: {
          email: 'E-mail Address'
        },
        custom: {
          name: {
            required: () => 'Name can not be empty',
            max: 'The name field may not be greater than 10 characters'
            // custom messages
          },
          select: {
            required: 'Select field is required'
          }
        }
      }
    }),

    mounted () {
      this.$validator.localize('en', this.dictionary)
    },

    methods: {
      submit () {
        var judge
        this.$validator.validateAll().then((result) => {
          if(!result) {
            alert(this.errors.all())
            return
          }
          this.registerUser()
        })
      },
      clear () {
        this.name = ''
        this.email = ''
        this.password = ''
        this.$validator.reset()
      },
      registerUser() {
        var params = {
          name: this.name, 
          email: this.email, 
          password: this.password
          }
        this.$store.dispatch('registerUser', ['/users', params])
      }
    }
  }
</script>

<style>
.formx {
  max-width: 768px
}
</style>
