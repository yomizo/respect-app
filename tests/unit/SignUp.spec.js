import { shallowMount } from '@vue/test-utils'
import SignUp from '@/components/SignUp.vue'

describe('SignUp', () => {
  const wrapper = shallowMount(SignUp)

  it('is a Vue instance', () => {  
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})