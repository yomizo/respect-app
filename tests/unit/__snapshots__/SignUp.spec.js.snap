// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp is a Vue instance 1`] = `
<v-content
  class="blue-grey lighten-5"
>
  <v-container
    class="formx"
  >
    <v-card
      class="opacity grey darken-4 pa-2"
    >
      <v-container>
        <v-card-text
          class="headline white--text font-weight-bold"
        >
          SignUp
        </v-card-text>
         
        <form>
          <v-text-field
            aria-invalid="false"
            aria-required="true"
            color="pink accent-2"
            counter="10"
            dark=""
            data-vv-name="name"
            error-messages=""
            label="Name"
            required=""
          />
           
          <v-text-field
            aria-invalid="false"
            aria-required="true"
            color="pink accent-2"
            dark=""
            data-vv-name="email"
            error-messages=""
            label="E-mail"
            required=""
          />
           
          <v-text-field
            aria-invalid="false"
            aria-required="true"
            class=""
            color="pink accent-2"
            dark=""
            name="password"
            placeholder="Password"
            type="password"
          />
           
          <div
            class="text-xs-right"
          >
            <v-btn
              color="pink accent-2"
              dark=""
            >
              submit
            </v-btn>
             
            <v-btn
              color="pink accent-2"
              dark=""
            >
              clear
            </v-btn>
          </div>
        </form>
      </v-container>
    </v-card>
  </v-container>
</v-content>
`;
